components/Featured/Featured.js

```diff
@@ -1,28 +1,35 @@
-import kid from "../../public/kid.png";
+import { doc, getDoc, getDocs, limit, orderBy, query } from "firebase/firestore";
+
+import { docToPost, postCollection } from "@/firebase/utils/postUtils";
+import { docToUser, userCollection } from "@/firebase/utils/userUtils";
 import AuthorCard from "../AuthorCard/AuthorCard";
 import PostCard from "../PostCard/PostCard";
 import styles from "./Featured.module.css";
 
-// mock user data
-const fetchMockData = (slug) => ({
-  id: 1234,
-  authorAvatar: kid,
-  displayName: "trinh andy",
-  blog: "Bob's Law Blog",
-  email: "admin@trinhandandy.com",
-  authorDescription:
-    "When the tangerine moose first heard about the concept of 'orange is the new black,' he couldn't help but feel a little personally offended.",
-  authorLinks: ["x", "y", "z"],
-});
-const data = fetchMockData();
-// mock end
+// get latest post
+const fetchPost = async () => {
+  const postQuery = await query(postCollection, orderBy("createdAt"), limit(1));
+  const postDoc = (await getDocs(postQuery)).docs;
+  return docToPost(postDoc[0]);
+};
+
+// get user
+const fetchUser = async (userId) => {
+  const userRef = await doc(userCollection, userId);
+  const userDoc = await getDoc(userRef);
+  return docToUser(userDoc);
+};
 
-const Featured = (props) => {
+const Featured = async (props) => {
+  const article = await fetchPost();
+  const user = await fetchUser(article.userId);
+  console.log(user);
+  console.log(user.blogSlug);
   return (
     <div className={styles.root}>
       <h2>FEATURED ARTICLE</h2>
-      <PostCard />
-      <AuthorCard {...data} futureStyle={styles.author} />
+      <PostCard {...article} {...user} />
+      <AuthorCard futureStyle={styles.author} {...user} />
```

Comment:
`futureStyle={styles.author}` is this meant to be className?

Reply:
you can do `className={styles.author}` then in AuthorCard component where you use that style ``<div className={`${style.root} ${props.className}`}>``

Reply:
ok ill test it out